app/views/layouts/application.html.erb

```diff
@@ -10,6 +10,58 @@
   </head>
 
   <body>
-    <%= yield %>
+    <nav class="navbar navbar-default">
+      <div class="container-fluid">
+        <div class="navbar-header">
+          <%= link_to root_path, class: "navbar-brand" do %>
+            <%= image_tag "logo" %>
+          <% end %>
+        </div>
+
+        <ul class="nav navbar-nav navbar-right">
+          <% if current_user.present? %>
+            <li class="dropdown">
+              <%= link_to "#", class: "dropdown-toggle", data: { toggle: "dropdown" }, role: "button", aria: { haspopup: "true", expanded: "false" } do %>
+                <%= fa_icon "bars" %>
+              <% end %>
+              <ul class="dropdown-menu">
+                <% if policy(:step).show_stage_1? %>
```

Comment:
These calls leverage what is referred to as a "Headless Policy" in Pundit, a Ruby authorization dependency that is great for decoupling logic for "who is allowed to do what", making it easy to separate the "who", "allowed", and "do what" parts without them all having to know about each other.